csv-bulk-signature/src/main.ts

```diff
@@ -41,85 +39,81 @@ interface TemplateData {
 const skippedRows: string[] = [];
 const processedRows: string[] = [];
 
-// create ./dist/signatures if it doesn't exist
-if (!fs.existsSync("./dist")) {
-  fs.mkdirSync("./dist");
-}
-if (!fs.existsSync("./dist/signatures")) {
-  fs.mkdirSync("./dist/signatures");
+async function setupFolders() {
+  // create ./dist/signatures if it doesn't exist
+  if (!fs.existsSync("./dist")) {
+    await fs.promises.mkdir("./dist");
+  }
+  if (!fs.existsSync("./dist/signatures")) {
+    await fs.promises.mkdir("./dist/signatures");
+  }
 }
 
-fs.readFile(TEMPLATE, "utf8", async (err, og_template) => {
-  if (err) throw err;
+async function getTemplate() {
+  const og_template = await fs.promises.readFile(TEMPLATE, "utf8");
 
   const inlinedTemplate = await inlineCss(og_template, {
     url: "./",
     removeHtmlSelectors: true,
   });
 
   // Compile the template
-  const template = await handlebars.compile<TemplateData>(inlinedTemplate);
-
-  // deleteAllSignatures();
-  checkHeadersToBeSame();
-  generateSignatures(template);
-});
+  return handlebars.compile<TemplateData>(inlinedTemplate);
+}
```

Comment:
wrap template setup in async funcation and move inner function calls to a main()